doc/explanation/storage.md

```diff
@@ -18,6 +18,9 @@ The following storage drivers are supported:
 - {ref}`storage-ceph`
 - {ref}`storage-cephfs`
 
+During initialization, you can create either a directory-based storage pool or a ZFS storage pool.
```

Comment:
I'm not sure thats true, will need to check...

Reply:
Darn, I checked whether it was "can" or "must", but stopped right after that (and released the VM).

Reply:
Confirmed:

```
lxd init
Would you like to use LXD clustering? (yes/no) [default=no]: 
Do you want to configure a new storage pool? (yes/no) [default=yes]: 
Name of the new storage pool [default=default]: 
Name of the storage backend to use (ceph, btrfs, dir, lvm, zfs) [default=zfs]: 
```

Reply:
`lxd init` lets you use all the different types except for cephfs.

```
stgraber@dakara:~/data/code/lxc/linuxcontainers.org (lxc/master)$ lxd init
Would you like to use LXD clustering? (yes/no) [default=no]: 
Do you want to configure a new storage pool? (yes/no) [default=yes]: 
Name of the new storage pool [default=default]: blah
Name of the storage backend to use (zfs, ceph, btrfs, dir, lvm) [default=zfs]: 
```

The logic for the default is somewhat complex but roughly it is:
 - Use zfs if the system is using zfs for its root
 - Use btrfs if the system is using btrfs for its root
 - Use zfs loop if zfs kernel support is detected
 - Use the dir backend

Reply:
Thanks for checking! Updated.